### ISSUE TITLE
Hold keys (continued activation)

### ISSUE THREAD

**Issue by f0n1x** [RaulPPelaez/Naga_KeypadMapper#42]
First off, great work! I appreciate being able to rebind my keys on my naga 2014.

I have rebound my keys on my pad and everything works great. But I use mumble for voice chat and i like to activate it via one of the keys that I have rebound on the pad (1 - key=grave) but the activation is instant and it stops. It's not registering that I am continuing to hold the key down. I can verify this by opening a text editor or a search field in a web browser and when I press and hold key 1 on the pad, I get a single ' (grave). Am I missing a setting, or is there a way for the program to register that I am keeping the button held down? TIA.

**Comment by TheGuardianWolf** [comment]
The program doesn't currently support hold actions (or at least nothing I see in the source indicates that)

**Comment by TheGuardianWolf** [comment]
As a sidenote, I can pull to put in a toggle function potentially, but press and hold detection requires a restructuring of the program.

**Comment by apocatarsis** [comment]
@f0n1x Thank you!
@TheGuardianWolf  You are right, the  code is thought in such a manner that only instant press is possible. But everything is done with xdotool, so I think it is pretty easy to add this feature.
xdotool has the option "xdotool keydown/keyup [key]" (currently the flag key is used). So storing the current state of each key (up/down) and using this xdotool flag will do the trick probably.

Nice!

**Comment by RaulPPelaez** [comment]
I merged your pull request @TheGuardianWolf, it did not compile as is but I think it should now be integrated with everything. Thank you very much!

It realized that this is actually simple to do even with the current code. The events read by the "select" function contain the type (which in this case will always be EV_KEY), and the code (1 for pressing, 0 for releasing). So the key operator should do keydown when this code is 1 and keyup when is 0. 
I pushed a commit doing this. Now the key operator does exactly what OP asked, keep the key pressed until the side button is released. Just as pressing it in the keyboard.

Thank you both!
